File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>> anagmap;

        for(auto& st: strs){
            string sortedst = st;
            sort(sortedst.begin(), sortedst.end());
            anagmap[sortedst].push_back(st);
        }

        vector<vector<string>> ans;
        for(auto& pair: anagmap){
            vector<string> anags = pair.second;

            ans.push_back(anags);
        }
        return ans;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: list[str]) -> list[list[str]]:
        hashmap = {}
        for word in strs:
            freq = [0]*26

            for char in word:
                idx = ord(char) - ord('a')
                freq[idx] += 1
            
            
            tup = tuple(freq)
            if tup in hashmap:
                hashmap[tup].append(word)

            else:
                hashmap[tup] = [word]
            
        return list(hashmap.values())

        